src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -359,6 +360,121 @@ public static void SendRecv_Stream_TCP(IPAddress listenAt, bool useMultipleBuffe
         [OuterLoop] // TODO: Issue #11345
         [Theory]
         [MemberData(nameof(LoopbacksAndBuffers))]
+        public static async Task SendRecvTask_Stream_TCP(IPAddress listenAt, bool useMultipleBuffers)
+        {
+            const int BytesToSend = 123456, ListenBacklog = 1, LingerTime = 1;
+            int bytesReceived = 0, bytesSent = 0;
+            Fletcher32 receivedChecksum = new Fletcher32(), sentChecksum = new Fletcher32();
+
+            var server = new Socket(listenAt.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+            server.BindToAnonymousPort(listenAt);
+            EndPoint clientEndpoint = server.LocalEndPoint;
+
+            server.Listen(ListenBacklog);
+            Task<Socket> serverAcceptTask = server.AcceptAsync();
```

Comment:
Why do the AcceptAsync, and then the WhenAll below, instead of just doing the AcceptAsync inside the server's Task.Run below?  Seems like that would make the logic more linear here, and therefore easier to read.